### ISSUE TITLE
Issues importing sleap data

### ISSUE THREAD

**Issue by IsabelleSajonia** [sgoldenlab/simba#284]
![error](https://github.com/sgoldenlab/simba/assets/70863857/68181865-b5cc-4ea5-9503-7bc070421d32)
Hi, I'm trying to start a new project and import labels from sleap. I tried importing as a .slp file and a H5 file I had. Neither worked and I get the following message. The video and predictions from sleap match, and I put both of these into a new file that just has what I want to import into simba. Any ideas what is wrong? Also why use a .slp file vs H5? I see in the sleap tutorial they say to export predictions as a H5.

**Comment by IsabelleSajonia** [comment]
Unrelated but I had another question about labeling- when I'm creating a new project there does not seem to be a function to zoom in when I'm labeling body parts on my mouse. We record in somewhat large operant chambers, so the mouse is small in the frame, and this makes the body part nodes much too large to accurately label in Simba.

**Comment by sronilsson** [comment]
Hi @IsabelleSajonia ! Thanks for posting the screengrab. The video you imported seems to be called `eating_3g_pellet_5min` (.mp4 or .avi..), but the data file you import seems to be called `Sleap_test_08172023.000_eating 3g pellet 5min.analysis`. SimBA needs a way of telling which data files goes with which video files, if you rename the files to have the same name, how does it look?

**Comment by sronilsson** [comment]
When you create a new body-part configuration in SimBA, the image is used for visualization purposes only - it is an image to pair with your pose-estimation body-part schema to show in the menus. It is not used for any calculations, so you do not have to be exact, you can click anywhere!

**Comment by IsabelleSajonia** [comment]
Ah yes changing the names to match fixed this! And the visualization makes sense I just wanted to make sure this wouldn't cause issues. Thank you.

**Comment by sronilsson** [comment]
Got it - Before extracting features, SimBA wants you to indicate how or if you want to correct outliers:

<img width="656" alt="image" src="https://github.com/sgoldenlab/simba/assets/34761092/7b384110-b86e-4564-97bd-eafbf706d3c6">

And documented [HERE](https://github.com/sgoldenlab/simba/blob/master/docs/Scenario1.md#step-4-outlier-correction) in Step 4. Once you have performed (or indicated to skip outlier correction), the directory `project_folder/csv/outlier_corrected_movement_location` will be populated with CSV files where the outliers are corrected (or skipped if clicked `SKIP`). Before you do this step, the `project_folder/csv/outlier_corrected_movement_location`  is empty. The feature extraction looks inside of the `project_folder/csv/outlier_corrected_movement_location` directory for files to extract features for, in your case it seems to find none.

**Comment by IsabelleSajonia** [comment]
No, this directory is empty even though I did run outlier correction and got the "complete" message.

**Comment by IsabelleSajonia** [comment]
I'm trying outlier correction again now and I get the messages "outlier correction settings updated" and "log for corrected "movement outliers" saved in project_folder/logs. Looking in my logs folder, I do see these csv files but they are empty (no data) other than the column headers (video, animal, body-part, etc.).

**Comment by sronilsson** [comment]
When you fix outliers, it is a two step process: first simba looks for any movement outliers and corrects them. Second, SimBA looks for a any location outliers and corrects them. The second part of is noticeably slower, especially if you have a lot of outliers, you can see the elapsed times:

<img width="799" alt="image" src="https://github.com/sgoldenlab/simba/assets/34761092/b50dbee6-b9cf-43b5-a3b0-f48471cdd5ad">


In your screengrab, it looks like the only outlier process that ran was the first step (movement outliers). If you check the main Windows terminal, do you see any error msg being printed after the movement outliers are complete?

**Comment by IsabelleSajonia** [comment]
![image](https://github.com/sgoldenlab/simba/assets/70863857/b26451f1-6d19-42c7-9b23-332fec0d9c0b)
I see this error in my terminal.

**Comment by IsabelleSajonia** [comment]
I do not see anything in that directory

**Comment by IsabelleSajonia** [comment]
Ok yes I think that is the issue I'm trying to import my H5 again and I get the following: 
![image](https://github.com/sgoldenlab/simba/assets/70863857/6faffa68-5cba-48fa-bc90-3e895ea221de)

I am selecting a folder on my desktop that just has the H5 file and the video clip that goes with it (both named the same thing).

**Comment by IsabelleSajonia** [comment]
![image](https://github.com/sgoldenlab/simba/assets/70863857/55f2ed23-6b80-434c-9772-7c0c295b336a)
Ok I tried the .slp file (and confirmed there are predictions in sleap) and now it is going through the frames but I get this error when it reaches the last one.

**Comment by sronilsson** [comment]
Almost there! :) 

When you selected the type of sleap file you are importing, did you select the `.SLP` kind from the dropdown? 

![image](https://github.com/sgoldenlab/simba/assets/34761092/c287144b-1046-495a-9b69-e9973c51bbcb)

**Comment by sronilsson** [comment]
Intresting, can you help with two things - if you do `pip show simba-uw-tf-dev` in the Windows terminal, which version number do you see? 

Is there any way you can share the `h5` file (not the slp file) it failed with first, and I can take a look whats going on? Maybe through a gdrive or drop it here in the chat if compressed to a zip file?

**Comment by IsabelleSajonia** [comment]
[eating_3g_pellet_5_min.zip](https://github.com/sgoldenlab/simba/files/12488937/eating_3g_pellet_5_min.zip)
[eating_3g_pellet_5_min (2).zip](https://github.com/sgoldenlab/simba/files/12488938/eating_3g_pellet_5_min.2.zip)
The first is the compressed h5 file and the second is the .slp file. And I'm seeing that its version 1.71.6

**Comment by sronilsson** [comment]
Thanks! I will take a look - in the meantime, you could try and update simba with `pip install simba uw-tf-dev --upgrade`, I tink latest version is 1.72.2, to confirm that you also see the same error.

**Comment by IsabelleSajonia** [comment]
I get the error "could not find a version that satisfies the requirement" with that line. For reference I installed simba a week or two ago following the anaconda installation doc

**Comment by sronilsson** [comment]
Do you have a screengrab or that error msg?

**Comment by IsabelleSajonia** [comment]
![error2](https://github.com/sgoldenlab/simba/assets/70863857/840628c3-0599-4bb7-aace-65a259478ea9)

**Comment by sronilsson** [comment]
The error comes from SLEAP normally having an entry called `track_names`, for whatever reason, your slp file has no track names. I haven't seen this before.. so we have to insert some code to handle that case. 

Can I also ask for the SLP file?

**Comment by sronilsson** [comment]
... and just fyi watch out for your last body part (tail_3?), it doesn't seem to be tracked very well a lot of missing values.

**Comment by IsabelleSajonia** [comment]
I think the second zipped file I sent you is a .slp file with predictions. And yes I'm having issues tracking the tail! Our operant chamber has a white floor and it blends in with the tip of this mouse's tail (I struggle to label it myself) so I may need to rethink my skeleton

**Comment by sronilsson** [comment]
Ah yes of course, let me just make sure it works with that file too and then I will give you updated code.

**Comment by sronilsson** [comment]
If you run `pip install simba-uw-tf-dev --upgrade`, giving you version `1.72.3`, how does it look on your end?

**Comment by IsabelleSajonia** [comment]
I just updated and used the H5 file and its working now! I can see that I have results in the features_extracted folder's csv. Thanks so much I really appreciate the help!

**Comment by sronilsson** [comment]
No problem and thanks! I wouldn't have known that the sleap h5 file could look like that if you hadn't let me know and shared the file.

**Comment by IsabelleSajonia** [comment]
Hopefully one last error- when I go to label behavior I get the following message: 

![Screenshot 2023-09-01 153414](https://github.com/sgoldenlab/simba/assets/70863857/90e9d573-0624-43b7-9c18-23d4517f8e97)

Could this have to do with tracking issues or is this related to trying to import the video itself? I'm seeing that the csv are now filled so I think everything up until this step is working ok now.

**Comment by sronilsson** [comment]
Thanks @IsabelleSajonia - no, it's a logging bug I must have introduced earlier in the week. If you update simba again with `pip install simba-uw-tf-dev --upgrade`, how does it look?

**Comment by IsabelleSajonia** [comment]
That fixed it I'm able to open the annotation interface now!

**Comment by sronilsson** [comment]
Excellent, thanks for letting me know. It was a typo of mine including [THIS](https://github.com/sgoldenlab/simba/issues/281#issuecomment-1703264587) for another user
